tutorials/interoperability/using-axelar-sdk.md

```diff
@@ -0,0 +1,226 @@
+
+In this case, since the origin chain is Moonbase Alpha, the native currency is DEV. We can use native DEV to pay for gas on the destination chain based on the conversion rates between Moonbase Alpha’s native currency and the destination chain’s native currency. Since we’re sending a contract call that includes a token to pay for destination gas in DEV, we will be using the `payNativeGasForContractCallWithToken` function.
+
+Finally, we call the gateway to send our cross-chain message with `callContractWithToken.` Notice that the payload (generic data that sent in a cross-chain call) that we’re sending is just the caller’s address. This data will need to be decoded by the destination contract.
+
+Now let’s take a look at what happens on the destination chain. Since we expect tokens to be sent as payment for an NFT mint, we will override `_executeWithToken` from `IAxelarExecutable.`
+
+???+ code "executeWithToken function"
+
+    ```solidity
+    --8<-- 'code/tutorials/interoperability/axelar-sdk/executeWithToken.sol'
+    ```
+
+In our implementation of `_executeWithToken`, we first check to ensure that the `tokenSymbol` provided by Axelar is “WDEV”. Then we expect 0.05 WDEV tokens for payment and will revert if any other token or anything less than 0.05 WDEV gets sent. Afterwards, we decode the payload to get the address of the origin chain’s caller so that we can mint an NFT to that address. Finally, we finish the minting!
+
+You can find the full code for the `CrossChainNFT.sol` below.
+
+??? code "CrossChainNFT.sol"
+
+    ```solidity
+    --8<-- 'code/tutorials/interoperability/axelar-sdk/CrossChainNFT.sol'
+    ```
+
+## Setting Up the Repository {: #setting-up-the-repository} 
+
+Make sure to clone the [GitHub repository](https://github.com/jboetticher/axelar-sdk-demo/tree/main){target=\_blank} for this tutorial. We need to install some dependencies, including Hardhat, OpenZeppelin contracts, some Axelar contracts, and the Axelar SDK. To configure the dependencies properly, run the following command:
+
+```bash
+npm install
+```
+
+The repository contains two Solidity files. The first file is the CrossChainNFT as expected, and the second is an Axelar library `StringAddressUtils.sol` that doesn’t have an npm package yet but is still required for the Hardhat implementation.
```

Comment:
```suggestion
The repository contains two Solidity files. The first file is the `CrossChainNFT` as expected, and the second is an Axelar library `StringAddressUtils.sol` that doesn’t have an npm package yet but is still required for the Hardhat implementation.
```

Either Cross Chain NFT contract or `CrossChainNFT` - sorry if I missed this one before! Please double-check all instances of CrossChainNFT

Reply:
done!